### ISSUE TITLE
Multiple schedulers should not be scheduling jobs at the same time

### ISSUE THREAD

**Issue by lmars** [flynn/flynn#1182]
In [this CI failure](https://s3.amazonaws.com/flynn-ci-logs/20150306025250-28bb2051-build-4dd2c56c89579d626cef5f0d0eca70f3e1530877-2015-03-06-03-03-58.txt), the leader scheduler (job ID `dbba4d53-ef7c6b6930b547138122509043ef95db`) has a heartbeat error at `02:59:45`:

```
2015/03/06 02:59:45 discoverd: heartbeat flynn-controller-scheduler (100.100.87.44:) failed: unknown_error: Something went wrong
```

A follower scheduler (job ID `04c7a5fc-61389b0609e24342b18f7fee7755cdd4`) is promoted to leader at `02:59:44` and begins scheduling jobs:

```
now=2015-03-06T02:59:44+0000 app=controller-scheduler at=leader
```

But the original leader continues to schedule jobs, so both rectify the same formation, leading to double the amount of jobs, and ultimately some test failures:

```
# from job ID 04c7a5fc-61389b0609e24342b18f7fee7755cdd4
now=2015-03-06T02:59:51+0000 app.id=5320598ba220412d9e9ce56415b82786 release.id=efd15c6a929a46409807a456c1c9805e at=new fn=watchFormations app=controller-scheduler
now=2015-03-06T02:59:51+0000 diff=1 at=update app=controller-scheduler app.id=5320598ba220412d9e9ce56415b82786 release.id=efd15c6a929a46409807a456c1c9805e type=crasher expected=1 actual=0 fn=rectify

# from job ID dbba4d53-ef7c6b6930b547138122509043ef95db
now=2015-03-06T02:59:51+0000 app=controller-scheduler fn=watchFormations app.id=5320598ba220412d9e9ce56415b82786 release.id=efd15c6a929a46409807a456c1c9805e at=new
now=2015-03-06T02:59:51+0000 app=controller-scheduler diff=1 at=update type=crasher expected=1 app.id=5320598ba220412d9e9ce56415b82786 release.id=efd15c6a929a46409807a456c1c9805e fn=rectify actual=0
```

The original leader should realise it is no longer the leader and stop scheduling jobs.

**Comment by lmars** [comment]
Another example: https://ci.flynn.io/builds/20150314172535-17cdc78e

Job `d66471ff-fdfc6f5bd5cb4832b8db310692011d3b` is initial leader then `aa061c74-2c861d3f98a940c8b424780a6bca2e86` becomes leader at `17:32:27`.

**Comment by lmars** [comment]
A user in IRC (@jzila) has been having repeated issues with this, at one point all three of his schedulers were scheduling jobs (so scaling something to 3 would start 9 jobs).

A quick fix to stop multiple schedulers reacting to formation changes is to stop the `watchFormations` loop when demotion happens, which looks like it will work fine but it reveals another issue.

All schedulers currently watch job events, with only the leader restarting crashed jobs, but the reason for this is that the follower schedulers don't have the cluster state in memory (because they haven't called [syncCluster](https://github.com/flynn/flynn/blob/master/controller/scheduler/main.go#L265)), so [this continue](https://github.com/flynn/flynn/blob/master/controller/scheduler/main.go#L461) always happens. Once a leader is demoted, it needs to both stop watching formations, and stop restarting jobs.

Putting this here as a reminder in case we decide to fix this before a much needed bigger refactor.

**Comment by lmars** [comment]
Another example of this:

https://s3.amazonaws.com/flynn-ci-logs/20150412061213-1f87029b-build-7f1e3c9504c7bb0e1655dcc6d1ab6032fe361fdb-2015-04-12-06-35-23.txt

**Comment by jzila** [comment]
More example CI failures: https://gist.github.com/jzila/b5e000fb16aeebb02a9a

**Comment by titanous** [comment]
This should be resolved by the new scheduler (#1914).
